ksqldb-rest-app/src/main/java/io/confluent/ksql/api/impl/QueryEndpoint.java

```diff
@@ -218,6 +224,7 @@ private QueryPublisher createPullQueryPublisher(
         metrics.recordRowsProcessed(
             Optional.ofNullable(r).map(PullQueryResult::getTotalRowsProcessed).orElse(0L),
             sourceType, planType, routingNodeType);
+        pullBandRateLimiter.add(responseBytes);
```

Comment:
We would need to call `system time` to record metrics above, maybe we can just get one system time once to save the number of currentSystimeMillis?

Reply:
Ditto for other `add` calls where we would potentially get `systemTime` multiple times.

Reply:
Keeping the logic of getting the `systemTime` internal to the `pullBandRateLimiter` seemed like a cleaner design. That way the common method interface is simpler and someone cannot pass in a bad value. It does come at the expense of potentially multiple redundant function calls. 

It is also an attempt at decoupling the metrics reporting logic and bandwidth limiting logic. We are already piggy-backing on the metrics reporting framework for limiting the bandwidth, and I did not want further coupling between the two.